### ISSUE TITLE
Add additional customization to Pending_Animation.svelte

### ISSUE THREAD

**Issue by greatbacon** [feltcoop/felt#6]
> This component could use a number of improvements, like customizing the delay. I'm curious what minimal API improvements would look like.

We could customize the char used for the waiting animation (like say a 💚 or 🌰 ), maybe the number of chars used in it. Definitely some cool opportunity here!

_Originally posted by @greatbacon in https://github.com/feltcoop/felt-ui/issues/5#issuecomment-811393666_

**Comment by ryanatkn** [comment]
(collecting ideas in order of rough priority)

1. custom content (string? slot?)
1. custom item count
1. custom delay
1. customize just the color (currently implemented with `--primary_color`)
1. custom animation

I think this is a good opportunity to golf on any particular implementations and discuss the minutia. (self-conscious bikeshedding in a low stakes example) Each of the above could be solved in several ways, with all sorts of tradeoffs. Svelte REPLs are easy to fork, so we can compare several versions side by side.

**Comment by greatbacon** [comment]
Maybe `PendingAnimation` would be a better name? I think `WaitingAnimation` conveys the thought pretty well though.

I've never heard the term `to golf` used in software dev? Do you have a link to context/etymology?

I'm down to do some bikshedding here! This is actually kind of fun lol

I think using props will help us keep our UI components more compartmentalized. Like rather than creating an expectation of downstream users needing (or getting) to inject a bunch of HTML junk into the component, there can be a clearly defined interface of what can (and thus can't) be tweaked.

As for 💚 & 🌰 , weren't you working on a Unicode library project for just that reason?

**Comment by ryanatkn** [comment]
> `<PendingAnimation />`

I like this better because of the `AsyncStatus` type in Gro, which is `'initial' | 'pending' | 'success' | 'failure'`. It matches with more modern Promise-y APIs than "waiting" for sure.

>weren't you working on a Unicode library project for just that reason?

mm yeah. The data is all there, ready to be carved. We can think from this component perspective: what's the API that would be lightweight and convenient for users of this `<PendingAnimation content={getSomeUnicodeThing()} />`?

Or even more high level: what workflow do we go with today? And what workflow might we think about developing in the future?

Today: we could get the Unicode project online, and use it to browse, copy the symbol directly whenever we use it. Some UX things in that project are broken but I could probably fix the worst quickly if I think it's worth it, and I believe it's deployable.

Also: we could have a little module here in `felt-ui`. I'm thinking `src/vocab/symbols.ts`.

```ts
// src/vocab/symbols.ts
export const felt = '💚';
export const gro = '🌰';
```

**Comment by ryanatkn** [comment]
> I think using props will help us keep our UI components more compartmentalized. Like rather than creating an expectation of downstream users needing (or getting) to inject a bunch of HTML junk into the component, there can be a clearly defined interface of what can (and thus can't) be tweaked.

I like the idea of constraining this interface to one `string` prop. It's simpler, and we will then treat any more complex use cases with a more complex component, most likely.

So that implies if we export a string prop for the contents, we could export a `color` prop directly as well, because semantically we know we have text that will respond to it. But should we? This is actually a very big question that I have partially explored in some private prototypes that I plan to port into this project. This is a clear-cut example use case. We currently have it respond to `--primary_color`: I'm open to even removing that.

**Comment by greatbacon** [comment]
Oooh, I do like the quick `src/lib/icons.ts` implementation. That seems like the shortest path to get what we want today. Maybe once we get to the point where we're looking at 100+ lines in the library we can rethink/revisit the Unicode project?

> 
> > code golf
> 
> [see Wikipedia on code golf](https://en.wikipedia.org/wiki/Code_golf) - normally it's in the context of getting it smaller and smaller. Here I'm thinking of similar but with "what's the best way to add feature X?"

Oh that explains why I've never heard of it; it's a game antithetical to my traditional programming ideology & ecosystem 😆 

Regarding CSS & the color styling, I'm not sure enough of my foundations to feel like I can weigh in on that decision. So I will defer to you on decisions there.

**Comment by ryanatkn** [comment]
Was listening to a coding podcast (<https://www.svelteradio.com/episodes/luke-edwards-the-man-with-many-libraries>) where the guest is one of the main authors of our dependencies, which are among the fastest and smallest in the land. He described his module authoring being like figuring out a puzzle. I like this framing of puzzling out the best design much more than "golfing".

**Comment by ryanatkn** [comment]
`Pending_Animation.svelte` is now our first published Svelte component, added in #91
